tc/sgx/trusted_worker_manager/enclave/work_order_processor_kme.cpp

```diff
@@ -0,0 +1,140 @@
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "enclave_t.h"
+#include <algorithm>
+#include <vector>
+#include <string>
+
+#include "error.h"
+#include "tcf_error.h"
+#include "types.h"
+
+#include "hex_string.h"
+#include "utils.h"
+#include "enclave_utils.h"
+
+#include "work_order_processor_kme.h"
+#include "workload_processor.h"
+
+namespace tcf {
+    WorkOrderProcessorKME::~WorkOrderProcessorKME() {
+        // Sanitize class members storing secrets
+        worker_encryption_key.clear();
+        session_key.clear();
+    }
+
+    /*
+     * Populates keys from work order requests into ExtWorkOrderInfoKME
+     * class members
+     *
+     * @param ext_wo_data - extended work order data which contains WPE's
+     *                      public encryption key
+     * @param ext_wo_info_kme - Instance of ExtWorkOrderInfoKME class
+     */
+    void WorkOrderProcessorKME::PopulateExtWorkOrderInfoData(
+            std::string ext_wo_data,
+            ExtWorkOrderInfoKME* ext_wo_info_kme) {
+
+            // Create work order data vector having index and
+            // decrypted work order keys
+            std::vector<tcf::WorkOrderData> in_wo_keys;
+            std::vector<tcf::WorkOrderData> out_wo_keys;
+            for (auto d : this->data_items_in) {
+                in_wo_keys.emplace_back(
+                    d.workorder_data.index, d.GetDataEncryptionKey());
+            }
+
+            for (auto d : this->data_items_out) {
+                out_wo_keys.emplace_back(
+                    d.workorder_data.index, d.GetDataEncryptionKey());
+            }
+
+            // Initializing ExtWorkOrderInfo member variables
+            // whose values to be used in pre-processing
+            ext_wo_info_kme->SetExtWorkOrderData(ext_wo_data);
+            ext_wo_info_kme->SetWorkOrderRequesterNonce(this->requester_nonce);
+            ext_wo_info_kme->SetWorkOrderSymmetricKey(this->session_key);
+            ext_wo_info_kme->SetWorkOrderInDataKeys(in_wo_keys);
+            ext_wo_info_kme->SetWorkOrderOutDataKeys(out_wo_keys);
+        }
+
+    /*
+     * Execute KME specific work order requests
+     *
+     * @param enclave_data - Instance of EnclaveData
+     */
+    std::vector<tcf::WorkOrderData> WorkOrderProcessorKME::ExecuteWorkOrder(
+        EnclaveData* enclave_data) {
+
+        std::vector<tcf::WorkOrderData> in_wo_data;
+        std::vector<tcf::WorkOrderData> out_wo_data;
+        if (data_items_in.size() > 0) {
+            // data_items_in vector is sorted based out of index
```

Comment:
I don;t think data_items_in is sorted based on index at this point. Either we should sort or remove the comments both here and `ExecuteWorkOrder` in WorkOrderProcessor.cpp

Reply:
removed the comment at both files